lib/smart_proxy_remote_execution_ssh/multiplexed_ssh_connection.rb

```diff
@@ -154,9 +158,11 @@ def establish_ssh_options
       ssh_options << "-o User=#{@ssh_user}"
       ssh_options << "-o Port=#{@ssh_port}" if @ssh_port
       ssh_options << "-o IdentityFile=#{@client_private_key_file}" if @client_private_key_file
+      ssh_options << "-o CertificateFile=#{@client_ca_cert_file}" if @use_ssh_certificates && @client_ca_cert_file
       ssh_options << "-o IdentitiesOnly=yes"
       ssh_options << "-o StrictHostKeyChecking=accept-new"
-      ssh_options << "-o UserKnownHostsFile=#{prepare_known_hosts}" if @host_public_key
+      ssh_options << "-o UserKnownHostsFile=#{@client_ca_known_hosts_file}" if @use_ssh_certificates && @client_ca_known_hosts_file
+      ssh_options << "-o UserKnownHostsFile=#{prepare_known_hosts}" if !@use_ssh_certificates && @host_public_key
```

Comment:
If `client_ca_cert_file` are `client_ca_known_hosts_file` are set, but the client doesn't accept them, will it fallback to traditional non-ca pubkey authentication?

Reply:
I will need to test this once I have the ssh cert setup in a functional state.

Reply:
It appears that the known hosts file does not need to exist at all, but the authentication seems to fail if the client does not accept the certificate. I will do some further testing, but I would say that authentication attempt failing upon not accepting the certificate is the desired outcome in regards to security.